pkg/workerapi/workerclient/client.go

```diff
@@ -0,0 +1,70 @@
+package workerclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	v1 "github.com/iver-wharf/wharf-cmd/api/workerapi/v1"
+	"google.golang.org/grpc"
+)
+
+// New creates a new client that can communicate with a Wharf worker server.
+//
+// Implements the Closer interface.
+func New(baseURL string, opts ClientOptions) (Client, error) {
+	rest, err := newRestClient(opts)
+	if err != nil {
+		return nil, err
+	}
+	return &client{
+		rest:    rest,
+		grpc:    newGRPCClient(baseURL, opts),
+		baseURL: baseURL,
+	}, nil
+}
+
+type client struct {
+	rest    *restClient
+	grpc    *grpcClient
+	baseURL string
+}
```

Comment:
Suggest just moving the ensure content of this file inside `workerclient.go` as they are so tightly coupled. Good to have the Client interface type declaration, constructor, and implementation in the same place as there's only one implementation.